lib/streamlit/metrics_util.py

```diff
@@ -0,0 +1,76 @@
+import os
+import platform
+import subprocess
+import threading
+import uuid
+
+from typing import Optional
+
+from streamlit import file_util
+
+def _get_machine_id_v1():
+    """Get the old machine id"""
+    if (
+        platform.system() == "Linux"
+        and os.path.isfile("/etc/machine-id") == False
+        and os.path.isfile("/var/lib/dbus/machine-id") == False
+    ):
+        print("Generate machine-id")
+        subprocess.run(["sudo", "dbus-uuidgen", "--ensure"])
```

Comment:
Running `sudo` feels like a red flag. Is it necessary to do so? What are the circumstances this will be run in? (Will anyone running Streamlit on Linux have this function called? What if they don't have `sudo` access?) - If sudo _is_ necessary, I think we really need inline docs here explaining why.

Reply:
This sudo currently exists in our codebase as I copied this code unchanged from `__init__.py`. One of the goals of this change is to use a new tracking_id for our metrics, while keeping a log of the previous one at the same time. 

To accurately log the previous id, we need to keep the sudo, since it's currently being generated as such. Ideally we wouldn't have had it in the first place. In some number of months we'll deprecate tracking of the old ID and this sudo will disappear along with those changes.